### PR TITLE
Adds built in functions $__interval_s, $__interval_m, and $__interval_h

### PR DESCRIPTION
**Pull request by rickard-von-essen**
This adds build in functions `$__interval_s`, `$__interval_m`, and `$__interval_h` similar to `$__interval_ms`.

These are convenient when doing calculations or scaling of a metric depending on the interval.

Example - to get _Requests per Seconds_ regardless of time range:
```
SELECT sum(m1_rate) / $__interval_s FROM resources WHERE $timeFilter GROUP BY time($interval)
```

### CODE REVIEW

**Comment by bergquist** [comment]
@torkelo I think this makes sense. Do you see any reason for nog merging this?

**Comment by torkelo** [comment]
Yes, feels very clunky to have this many and we need to add them to the backend tsdb layer as well. Maybe supporting seconds is reasonable as it's probably more common that you want adjust things per second, the others i think we do not need as much

File: public/app/features/panel/metrics_panel_ctrl.ts [hunk 1]
```diff
@@ -25,6 +25,9 @@ class MetricsPanelCtrl extends PanelCtrl {
   range: any;
   interval: any;
   intervalMs: any;
+  intervalS: any;
```
Comment (torkelo) [inline public/app/features/panel/metrics_panel_ctrl.ts]: Having all variants variants as instance properties feel bad

**Comment by rickard-von-essen** [comment]
> Maybe supporting seconds is reasonable as it's probably more common that you want adjust things per second, the others i think we do not need as much

I suggest that `m` is convenient when doing RPM etc, even though `s` will do. The most important is to get something nicer than `ms` forcing me to do thing cumbersome things like `$__interval_ms * 300000`.

On the other hand adding all of them don't add any maintenance or performance overhead so why not make it cleaner to write queries?

**Comment by rickard-von-essen** [comment]
> we need to add them to the backend tsdb layer as well.

I did some testing and the current implementation seems to work. Can you point me to where this needs to be added.

**Comment by torkelo** [comment]
closing this for now until we find a more generic solution like $__interval(ms)

**Comment by rickard-von-essen** [comment]
@torkelo Would you accept a PR adding `$__interval(<ms|s|m|h>)`?

**Comment by torkelo** [comment]
yes! it would involve a very big rewrite of the variable interpolation engine (frontend and backend) so support function argument syntax

**Pull request closed without merging**